### PR TITLE
Improve email font family and footer settings

### LINKED ISSUE #52222: Improve email font family and footer settings
**Behind** the experimental flag from #52212.

- Please also check [Figma file](https://href.li/?https://www.figma.com/design/FAA2sMXXfT8Qs40AC3V1gL/WC-Core-Emails?node-id=594-7764) for more details and different states.
- Add `Font family` setting:
  - Options: the same as MailPoet fonts (excluding Google Fonts)
    - For block-based themes, also include the theme's font
    - Default to Helvetica.
  - Tooltip text: none
- Add new variables that can be used in emails:
  - `{store_address}` - a one-line physical address taken from **WooCommerce > Settings > General > Store Address** (combined all fields, comma separated).  
  - `{store_email}` - a "From" address setting from WooCommerce > Settings > Emails 
- Improve `Footer text`:
  - Remove `Built with {WooCommerce}`
  - Add `{store_address}` on a new line
  - Tooltip text (final, proof-read): `This text will appear in the footer of all of your WooCommerce emails.`

Image

### PR DESCRIPTION
**Pull request by alex-mpoet**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

When the feature flag is enabled, a font family setting is added to email settings. Also, the footer text setting is improved with a new description, a default value and new placeholders: `{store_address}` and `{store_email}`.

Closes #52222.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to `WooCommerce > Settings > Emails` and observe the absence of the font family setting and an old footer text setting.
2. Enable the email_improvements feature flag. It's hidden in the UI, so this needs to be done in the database: set the `woocommerce_feature_email_improvements_enabled` option to `yes`.
3. Reload the email settings page.
4. You should now see the font family setting which allows to select standard fonts as well as fonts from theme.json.
5. You will also see that the email footer text setting is updated with a new description and default value.
5. New placeholders are added to the footer text: `{store_address}` and `{store_email}`. Try using them in an email and see that they are properly expanded.

### CODE REVIEW

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php
Comment (NeosinneR) [inline plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php]: What do you think about `ksort()`ing this array? From the user's perspective, even if the beginning is sorted (the email-safe fonts), the end of the list of fonts will discard that assumption once theme fonts are appended.

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php
Comment (NeosinneR) [inline plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php]: Is it possible for each select option to be rendered in the font family it's representing?
Screenshot 2024-11-28 at 10 30 05

In the site editor, you see how the font looks, which makes it much easier to select the font you like.
Screenshot 2024-11-28 at 10 29 56
Screenshot 2024-11-28 at 10 31 36

**Review by NeosinneR — COMMENTED** [review]
@alex-mpoet looks and works as expected. I only have two suggestions for improving the UX, but let me know what you think about them.

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php
Comment (alex-mpoet) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php]: It seems the stock Woo settings admin select component doesn't support this kind of customization, so I've added our own implementation of the field for that.

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php
Comment (alex-mpoet) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php]: We can do that for sure. Maybe a better way to separate standard and custom fonts would be to introduce optgroups like in MailPoet. What do you think about that?

Comment (NeosinneR) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php]: That's a good idea, I like it. I'll let you decide which one to pick.

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php
Comment (NeosinneR) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php]: @alex-mpoet could you please share how it looks for you? I see only the native OS select box, where styles are not applied.

Screenshot 2024-11-29 at 13 07 15

Comment (alex-mpoet) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php]: @NeosinneR You're right, that simple select-based solution had issues with rendering in Safari. I have implemented a select2-based one which is based on MailPoet's approach, it uses the `selectWoo` component already available on the Woo settings admin page. It now should render fine in Safari and other browsers.

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php
Comment (alex-mpoet) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php]: I have added optgroups, please check them out.

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php
Comment (NeosinneR) [inline plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php:657]: Many of the `$value` properties are not used in this method (maybe only `title`, `id`, and `default`?). As I understand it, this code is based on a template, which can accept many of these properties and render it differently, but we have full control over the email font families component. What do you think about simplifying it? Or is there a use case where it's useful to have it there?

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php
Comment (NeosinneR) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php]: The groups are a nice addition 👍 Just a small note, that the `Custom fonts` is empty when no custom fonts are available, which looks broken (as in the fonts haven't loaded). I'd probably just hide the group completely.

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php
Comment (alex-mpoet) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php:657]: I was thinking of future extensibility, such as "what if we want to add a tooltip here" etc. Plus, some of these properties are added automatically and not taken from the config array. But I agree that it's better to keep things simple, so I have cleaned it up wherever possible.

File: plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php
Comment (alex-mpoet) [reply plugins/woocommerce/includes/admin/settings/class-wc-settings-emails.php]: Nice catch! Fixed.

**Review by NeosinneR — APPROVED** [review]
Looks good, @alex-mpoet. Thanks for the updates. I pushed two changes to speed up review:
1. Fixed the wrong indentation (I'm surprised this wasn't caught by any linter).
2. Removed the font size in the select box, as it was too large.

Screenshot 2024-12-03 at 16 54 52

I'll wait for the build to pass before merging.

**Pull request merged**
